controllers/controllers/resource/fetcher.go

```diff
@@ -157,14 +157,25 @@ func (r *CapiResourceFetcher) fetchClusterForRef(ctx context.Context, refId type
 			}
 		}
 		if kind == anywherev1.AWSIamConfigKind {
-			for _, indentityProviderRef := range c.Spec.IdentityProviderRefs {
-				if indentityProviderRef.Name == refId.Name {
+			for _, identityProviderRef := range c.Spec.IdentityProviderRefs {
+				if identityProviderRef.Name == refId.Name {
 					if _, err := r.clusterByName(ctx, constants.EksaSystemNamespace, c.Name); err == nil { // further validates a capi cluster exists
 						return &c, nil
 					}
 				}
 			}
 		}
+
+		if kind == anywherev1.OIDCConfigKind {
```

Comment:
you can combine these two fetch with `if kind == anywherev1.OIDCConfigKind || kind == anywherev1.AWSIamConfigKind {}` since the code is the same